### ISSUE TITLE
About the data

### ISSUE THREAD

**Issue by 384068026** [CarrieX6/Self-supervised-EEG-classify#1]
Hi, I am trying to run your code, but I find that the data processing section calls the keywords 'traindata' and 'testdata' and indexes like 'trainlabel' directly. It seems to me that there is a missing data processing file from the original SEED data to the data required for the operation of your project. Did you not publish this part? Or am I doing something wrong?

**Comment by CarrieX6** [comment]
Hi there, did you download the original SEED dataset from the given website? I preprocessed the original data and split it into testdata and traindata matrices by the ratio of 4:6. Consider processing the data and put into model, please.

**Comment by 384068026** [comment]
Ok, at first I thought you were using the data from the Preprocessed_EEG folder under the SEED dataset.  After reading your comments about the shape of the data I realize that you should use the ExtractedFeatures folder.  You wrote traindata: (5, 62, 2010) and testdata: (5, 62, 1384) in the comments.  So you stack 15 people's data and then split the training set and the test set?  Take the data in de_movingAve1.  The shape of each cell is (62,235,5).  So for 15 people, 235*15=3525, divided by 6:4, the numbers are 2115 and 1410.  The number is close to your remarks but does not match.  Is it convenient to tell which feature you used?  Thank you very much for your help.

**Comment by CarrieX6** [comment]
Sorry, it's been too long to remember clearly which data folder I used. The input shape should be (number, 62, 5). After my preprocessing steps, there are 45 .mat files, and each file contains testdata(1384*62*5), testlabel(1*1384), traindata(2010*62*5), and trainlabel(1*2010). Cause I am not researching in this direction anymore, I didn't keep the flow of data preprocessing. I can send you the after-preprocessing data mat files if needed, but you may try some data directly with the same channel.

**Comment by CarrieX6** [comment]
Sorry, it's been too long to remember clearly which data folder I used. The input shape should be (number, 62, 5). After my preprocessing steps, there are 45 .mat files, and each file contains testdata(1384  \times 62  \times 5), testlabel(1 \times 1384), traindata(2010 \times 62 \times 5), and trainlabel(1 \times 2010). Cause I am not researching in this direction anymore, I didn't keep the flow of data preprocessing. I can send you the after-preprocessing data mat files if needed, but you may try some data directly with the same channel.

**Comment by CarrieX6** [comment]
Sorry, it's been too long to remember clearly which data folder I used. The input shape should be (number, 62, 5). After my preprocessing steps, there are 45 .mat files, and each file contains testdata(1384  \* 62  \* 5), testlabel(1 \* 1384), traindata(2010 \* 62 \* 5), and trainlabel(1 \* 2010). Cause I am not researching in this direction anymore, I didn't keep the flow of data preprocessing. I can send you the after-preprocessing data mat files if needed, but you may try some data directly with the same channel.

**Comment by 384068026** [comment]
Ok.Thank you very much for sending me a copy of the processed data so that I can study it carefully. My email is shichi1999js@163.com

**Comment by CarrieX6** [comment]
931706813
***@***.***



&nbsp;






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "CarrieX6/Self-supervised-EEG-classify"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月27日(星期一) 晚上9:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [CarrieX6/Self-supervised-EEG-classify] About the data (Issue #1)





 
Ok.Thank you very much for sending me a copy of the processed data so that I can study it carefully. My email is ***@***.***
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

SEED.zip (312.80M, 2023年04月26日 21:15 到期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?t=exs_ftn_download&k=0a3731615a2147db8a80b956406206184f4303505e07510f1a530050504f56020e561c58075a021a04020205500355530f540652666a34647272754f1c0b44370a&code=771afb47

**Comment by CarrieX6** [comment]
You are welcome. Already sent. Please make sure you have gained permission from http://bcmi.sjtu.edu.cn/~seed. And cite the SEED-associated articles when you use them in your paper.

**Comment by 384068026** [comment]
I have applied for the dataset and obtained authorization from the SEED administrator. And I understand the rules that need to be followed. Thank you very much for your help. Wish you a happy life and work.

**Comment by CarrieX6** [comment]
That's great. Thank you and good luck to your research.

**Comment by crlijiatu** [comment]
> de_movingAve1
Hello, I recently read your article and found the same problem when running the code. Could you please send me your pre-processed file? I have requested the dataset and obtained authorization from the SEED administrator.

**Comment by crlijiatu** [comment]
> > de_movingAve1 您好，我最近读了您的文章，在运行代码时发现了同样的问题。你能把你预处理的文件发给我吗？我已请求数据集并获得了 SEED 管理员的授权。

my email is cr0201021@email.swu.edu.cn
